### PR TITLE
statusbar: list developer settings in tooltip

### PR DESCRIPTION
**Pull request by justinmk3**
## Problem

Want to know which settings triggered developer mode.

## Solution

Show the settings in the statusbar tooltip.
Screen Shot 2021-09-22 at 5 46 46 PM

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/shared/awsContext.ts [hunk 1]
```diff
@@ -17,34 +20,143 @@ export interface AwsContextCredentials {
...
+}
+
+const logged = new Set<string>()
+const DEFAULT_REGION = 'us-east-1'
+
+/**
+ * Wraps an AWS context in terms of credential profile and zero or more regions. The
+ * context listens for configuration updates and resets the context accordingly.
+ */
+export class DefaultAwsContext implements AwsContext {
```
Comment (justinmk3) [inline src/shared/awsContext.ts:40]: moved from src/shared/defaultAwsContext.ts

File: src/shared/awsContext.ts [hunk 2]
```diff
@@ -17,34 +20,143 @@ export interface AwsContextCredentials {
...
+        this.emitEvent()
+    }
+
+    /**
+     * Sets "developer mode" when a Toolkit developer setting is active.
+     *
+     * @param enable  Set "developer mode" as enabled or disabled
+     * @param settingName  Name of the detected setting, or undefined for `enable=false`.
+     */
+    public async setDeveloperMode(enable: boolean, settingName: string | undefined): Promise<void> {
```
Comment (justinmk3) [inline src/shared/awsContext.ts:78]: this function is the only changed part

File: src/shared/awsContext.ts [hunk 3]
```diff
@@ -17,34 +20,143 @@ export interface AwsContextCredentials {
...
+            return
+        }
+
+        if (!enable) {
+            this.developerMode.clear()
+        } else {
+            this.developerMode.add(settingName ?? '?')
+        }
+        this.emitEvent()
+    }
```
Comment (justinmk3) [inline src/shared/awsContext.ts:91]: this is the only changed part

**Review by JadenSimon — APPROVED** [review]

**Pull request merged**